helix-view/src/editor.rs

```diff
@@ -1671,6 +1682,14 @@ impl Editor {
         &self,
         timeout: Option<u64>,
     ) -> Result<(), tokio::time::error::Elapsed> {
+        // Remove all language servers from the file event handler.
+        // Note: this is non-blocking.
+        for client in self.language_servers.iter_clients() {
+            self.language_servers
+                .file_event_handler
+                .remove_client(client.id());
```

Comment:
Does this clean-up need to be run in other places where clients are shutdown/removed like `Registry::shutdown`, `remove_by_id` and `restart`?

Reply:
Great point! Originally the `file_event::Handler` lived on the `Editor` struct, but now that it's part of the `Registry`, I think it's best to do the cleanup from those `Registry` methods themselves. I've added `remove_client` calls to the methods: "remove_by_id", "restart", and "stop".